src/main/navigation/navigation_fixedwing.c

```diff
@@ -136,7 +136,12 @@ static void updateAltitudeVelocityAndPitchController_FW(timeDelta_t deltaMicros)
 
     // PID controller to translate energy balance error [J] into pitch angle [decideg]
     float targetPitchAngle = navPidApply3(&posControl.pids.fw_alt, demSEB, estSEB, US2S(deltaMicros), minDiveDeciDeg, maxClimbDeciDeg, 0, pitchGainInv);
-    targetPitchAngle = pt1FilterApply4(&velzFilterState, targetPitchAngle, NAV_FW_PITCH_CUTOFF_FREQENCY_HZ, US2S(deltaMicros));
+
+    // Apply low-pass filter to prevent rapid correction cutoff_freq = 0.002 * (10.0-x)^3 + epsilon 
+    // Pitch cutoff set to 1/5 the roll cutoff
+    uint16_t smoothness = 10 - navConfig()->fw.control_smoothness;
+    float cutoff_freq = 0.002f * (float)(smoothness*smoothness*smoothness) + 0.1f;
+    targetPitchAngle = pt1FilterApply4(&velzFilterState, targetPitchAngle, cutoff_freq, US2S(deltaMicros));
```

Comment:
Simple math:

control_smoothness = 0 -> cutoff_freq = 2
control_smoothness = 5 -> cutoff_freq = 0.35
control_smoothness = 9 -> cutoff_freq = 0.102
control_smoothness = 10 -> cutoff_freq = 0.1

I'd say values above 5 would make no visible difference. Maybe use a `^2` curve or make cutoff frequencies directly configurable?

Reply:
Actually values above 5 start to make even more difference, hence the ^3 in an attempt to make it more progressive, without going to fully accurate math using divisions (maybe a foolish optimisation but still). It makes more difference above 5 because the smoothness feeling is analogous to the cutoff period and not the frequency. 

I didn't go with a direct frequency parameter here as I find it more natural for a regular user to use a 1-10 scale than having to understand frequencies. For example for a user to go from 1 second banking to 2 seconds banking he would have to intuitively understand that he should adjust the freq cutoff from ~1Hz to ~0.5Hz. Then for 3 seconds to ~0,33Hz.

Therefore if we don't like the generic smoothness value and want to expose to the user the low level mechanics of smoothing I would expose the cutoff period. But then we should again explain that a period of 2sec doesn't exactly mean the airplane will bank in 2sec, as the actual bank will depend on the actual plane make, weight, etc. That's why I like the generic unit-less smoothness setting. We can make the math more "correct" though for sure, if you think that is important.

Reply:
Makes sense. Can you make a unified function that will accept a base cutoff frequency in Hz instead of magic 0.002f here and use that function in both pitch and roll to avoid code duplication?

Reply:
Sure!